homeassistant/components/lutron_caseta/switch.py

```diff
@@ -25,9 +25,12 @@ async def async_setup_entry(
     data: LutronCasetaData = hass.data[CASETA_DOMAIN][config_entry.entry_id]
     bridge = data.bridge
     bridge_device = data.bridge_device
+    device_info_by_device_id = data.device_info_by_device_id
     switch_devices = bridge.get_devices_by_domain(DOMAIN)
     async_add_entities(
-        LutronCasetaLight(switch_device, bridge, bridge_device)
+        LutronCasetaLight(
+            switch_device, bridge, bridge_device, device_info_by_device_id
```

Comment:
I think this would be easier to maintain if we passed in a `data` here instead so we don't have to unpack it and feed it into every entity.

Reply:
Ok, no problem.  I've updated the the __init__ method to pass in the data variable instead of the bridge, bridge_device, and device_info_by_device_id parameters.